content/3_Silver/Sorting_Custom.mdx

```diff
@@ -48,626 +46,461 @@ After sorting, it should look like
 2 3 10
 ```
 
-With C++, the easiest method is to use a `vector` of nested `pair`s:
+<LanguageSection>
+<CPPSection>
+
+There's multiple ways we can do this.
+The most straightforward method is to use a `pair<int, pair<int, int>>`.
+When comparing these, C++ looks at the first element and only uses
+the second element as a tiebreaker.
 
 ```cpp
-#include <bits/stdc++.h>
-using namespace std;
+#include <algorithm>
+#include <iostream>
+#include <vector>
 
-#define f first
-#define s second
+using namespace std;
 
 int main() {
-	int M = 4;
-	vector<pair<int, pair<int, int>>> v;
-	for (int i = 0; i < M; ++i) {
-		int a, b, w;
-		cin >> a >> b >> w;
-		v.push_back({w, {a, b}});
-	}
-	sort(begin(v), end(v));
-	for (auto e : v) cout << e.s.f << " " << e.s.s << " " << e.f << "\n";
-}
-```
+	constexpr int edge_num = 4;
 
-or a `vector` of `array<int,3>`s or `vector<int>`s:
+	vector<pair<int, pair<int, int>>> edges(edge_num);
+	for (auto &[width, edge] : edges) { cin >> edge.first >> edge.second >> width; }
+	sort(edges.begin(), edges.end());
 
-```cpp
-int main() {
-	int M = 4;
-	vector<array<int, 3>> v;  // or vector<vector<int>>
-	for (int i = 0; i < M; ++i) {
-		int a, b, w;
-		cin >> a >> b >> w;
-		v.push_back({w, a, b});
+	for (const auto &[width, edge] : edges) {
+		printf("(%i, %i): %i\n", edge.first, edge.second, width);
```

Comment:
wait why printf?

Reply:
look at the formatting i give around it, cout would be hell